### PR TITLE
pasta: use code from c/common

### PR DESCRIPTION
**Pull request by Luap99**
The code was moved to c/common so use that instead. Also add tests for
the new pasta_options config field. However there is one outstanding
problem[1]: pasta rejects most options when set more than once. Thus it is
impossible to overwrite most of them on the cli. If we cannot fix this
in pasta I need to make further changes in c/common to dedup the
options.

[1] https://archives.passt.top/passt-dev/895dae7d-3e61-4ef7-829a-87966ab0bb3a@redhat.com/

#### Does this PR introduce a user-facing change?

```release-note
A default list of pasta(1) arguments can now be set in containers.conf via `pasta_options` options.
```

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

**Review by vrothberg — COMMENTED** [review]
LGTM

**Comment by Luap99** [comment]
Repushed, I had to add the skip_if_remote to the test.

File: libpod/networking_pasta_linux.go [hunk 1]
```diff
@@ -7,105 +7,13 @@
...
-import (
-	"fmt"
-	"os/exec"
-	"strings"
-
-	"github.com/sirupsen/logrus"
-)
-
-const (
-	pastaBinaryName = "passt"
```
Comment (edsantiago) [inline libpod/networking_pasta_linux.go:19]: The new libnetwork/pasta/pasta.go calls it `pasta` and is more consistent in that use (vs line 29 below). On my f38 laptop I see `/usr/bin/pasta -> passt`; also on f37. I'm assuming this is a deliberate change, and that the `passt` RPM will forevermore guarantee this symlink, but I don't like silent assumptions so I'm pointing it out.

Comment (Luap99) [reply libpod/networking_pasta_linux.go:19]: I made it an extra commit in c/common after moving the code: https://github.com/containers/common/pull/1506/commits/d9ee72cee5eebc1fbe403e972bc89bc30fc59253
Yes it is always a symlink but we only ever use pasta so it doesn't make sense to call it passt in info IMO.

**Pull request merged**